public/app/features/alerting/unified/components/admin/ConfigEditor.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+
+import { Button, ConfirmModal, TextArea, HorizontalGroup, Field, Form } from '@grafana/ui';
+
+import { GRAFANA_RULES_SOURCE_NAME } from '../../utils/datasource';
+
+import { FormValues } from './AlertmanagerConfig';
+
+interface ConfigEditorProps {
+  defaultValues: { configJSON: string };
+  readOnly: boolean;
+  loading: boolean;
+  alertManagerSourceName?: string;
+  onSubmit: (values: FormValues) => void;
+  showConfirmDeleteAMConfig?: boolean;
+  onReset?: () => void;
+  onConfirmReset?: () => void;
+  onDismiss?: () => void;
+}
+
+export const ConfigEditor = ({
+  defaultValues,
+  readOnly,
+  loading,
+  alertManagerSourceName,
+  showConfirmDeleteAMConfig,
+  onSubmit,
+  onReset,
+  onConfirmReset,
+  onDismiss,
+}: ConfigEditorProps) => {
+  return (
+    <Form defaultValues={defaultValues} onSubmit={onSubmit} key={defaultValues.configJSON}>
+      {({ register, errors }) => (
+        <>
+          {!readOnly && (
```

Comment:
Maybe we could group this  repeated `{!readOnly &&` ?

Reply:
Good idea! Done